# HG changeset patch
# User Rik <rik@octave.org>
# Date 1461694564 25200
#      Tue Apr 26 11:16:04 2016 -0700
# Node ID 9810e483b42fa273a89ea82bd7abbabafef75413
# Parent  7bdfa98a98a30c5d71a3f74f057f091586f9b4b8
Fix dbstop BIST tests on Windows platforms.

* debug.cc: Change BIST tests to use filesep() which can depend on the
operating system, rather than a fixed string.

diff --git a/libinterp/corefcn/debug.cc b/libinterp/corefcn/debug.cc
--- a/libinterp/corefcn/debug.cc
+++ b/libinterp/corefcn/debug.cc
@@ -426,16 +426,17 @@ parse_dbfunction_params (const char *who
           default:      // dbstop_none should never occur
             break;
         }
     }
 }
 
 /*
 %!test
+%! dbclear all;   # Clear out breakpoints before test
 %! dbstop help;
 %! dbstop in ls;
 %! dbstop help at 100;
 %! dbstop in ls 100;
 %! dbstop help 200 if a==5;
 %! dbstop if error Octave:undefined-function
 %! s = dbstatus;
 %! dbclear all
@@ -1484,27 +1485,28 @@ The @qcode{\"warn\"} field is set simila
         }
 
       return retval;
     }
 }
 
 /*
 %!test
+%! dbclear all;   # Clear out breakpoints before test
 %! dbstop @ftp/dir;
 %! dbstop @audioplayer/set 70;
 %! dbstop quantile>__quantile__;
 %! dbstop ls;
 %! s = dbstatus;
 %! dbclear all
 %! assert (s(1).name, "@audioplayer/set>setproperty");
 %! assert (s(2).name, "@ftp/dir");
 %! assert (s(3).name, "ls");
 %! assert (s(4).name, "quantile>__quantile__");
-%! assert (s(2).file(end-10:end), "/@ftp/dir.m");
+%! assert (s(2).file(end-10:end), [filesep "@ftp" filesep "dir.m"]);
 */
 
 DEFUN (dbwhere, , ,
        "-*- texinfo -*-\n\
 @deftypefn {} {} dbwhere\n\
 In debugging mode, report the current file and line number where execution\n\
 is stopped.\n\
 @seealso{dbstack, dblist, dbstatus, dbcont, dbstep, dbup, dbdown}\n\
